enhancements/baremetal-IPI/baremetal.md

```diff
@@ -0,0 +1,220 @@
+
+## Motivation
+
+The motivation for this enhancement request is to provide a background for all the
+the subsequent enhancement requests for Baremetal IPI deployments.
+
+### Goals
+
+The goal of this enhancement request is to provide context for all the changes
+that have already been merged towards making Baremetal IPI deployments a reality.
+All future Baremetal enhancement requests will refer back to this one to provide
+context.
+
+### Non-Goals
+
+No development PRs will be raised as a result of this enhancement request.
+
+## Proposal
+
+Every OpenShift based Baremetal IPI deployment will run a "metal3" pod on
+all Master Nodes. A "metal3" pod includes a container running BareMetal
+Operator(BMO) and several other supporting containers that work togther.
+
+The BMO and other supporting containers together are able to discover a
+baremetal server in a pre-determined provisioning network, learn the
+HW attributes of the server and eventually boot it to make it available
+as a Machine within a MachineSet.
+
+The Machine API Operator currently deploys the "metal3" pod only when the
+Platform type is Baremetal. The MAO also installs the BaremetalHostCRD
+for every type of deployment starting from release 4.2.
+
+The installer is responsible for kicking off a Baremetal IPI deployment
+with the right configuration.
+
+A set of follow-up enhancement requests will try to bring in the following
+enhancements:
+1. Providing the BMO and it supporting containers with configuration that
+is specific to the server and its provisioning environment. This enhancement
+request is already under review here : 
+https://github.com/openshift/enhancements/pull/90
+
+2. The MAO would also be responsible for deploying a DeamonSet that would
+run on all baremetal servers to monitor the hardware inventory on each
+of these servers.
+
+### User Stories [optional]
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+The MAO already deploys the "metal3" pods with the BareMetal Operator and the
+other (Ironic) containers. Currently, there is no mechanism in place to provide
+these containers with their configuration.
+
+### Risks and Mitigations
+
+Will be specified in follow-up enhancement requests mentioned above.
+
+## Design Details
+
+### Test Plan
+
+True e2e and integration testing can happen only after implementation for
+enhancement https://github.com/openshift/enhancements/pull/90 lands.
+Until then, e2e testing is being performed with the help of some
+developer scripts.
+
+Unit tests have been added to MAO and the Installer to test additions
+made for the Baremetal IPI case.
+
+### Graduation Criteria
```

Comment:
Maybe in this section we should say that the metal3 integration is already Tech Preview for 4.2 and scheduled for GA in whatever release we've promised it will be.